sale_analytic_account/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        for so in self:
+            if any(so.order_line.filtered(
+                    lambda line: line.product_id.create_analytic_account)):
+                analytic_account_id = self.env['account.analytic.account'].\
+                    create({'name': so.name})
+                so.analytic_account_id = analytic_account_id
```

Comment:
```suggestion
                so.analytic_account_id = \
                    self.env['account.analytic.account'].create({
                        'name': so.name,
                    })
```

Reply:
Please also set the customer of the analytic account, to be the same as the customer of the sale.order
That's how it works for sale_timesheet I think